### ISSUE TITLE
AMD_LOG_MASK appears not working although it is documented well

### ISSUE THREAD

**Issue by jdgh000** [ROCm-Developer-Tools/HIP#3298]
Operating environment 
```
83:00.0 Display controller: Advanced Micro Devices, Inc. [AMD/ATI] Aldebaran (rev 02)
a3:00.0 Display controller: Advanced Micro Devices, Inc. [AMD/ATI] Aldebaran (rev 02)
root@u2204:~/root/ml# cat /etc/os-release
PRETTY_NAME="Ubuntu 22.04.2 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.2 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
root@u2204:~/root/ml# uname -r
5.19.0-50-generic
root@u2204:~/root/ml# cat /opt/rocm/.info/version
5.4.2-104
root@u2204:~/root/ml#
pip3 list | grep torch
pytorch-triton-rocm       2.0.1
torch                     2.0.1+rocm5.4.2
torchaudio                2.0.2+rocm5.4.2
torchdata                 0.6.1
torchtext                 0.15.2
torchvision               0.15.2+rocm5.4.2

```
tried several flags with ML tasks but regardless of what value given to AMD_LOG_MASK, it always seem to produce same output.
Is that right way to use the flag? Under ROCClr, i see there is a limited application of this environment variable AMD_LOG_MASK but just does not seem to be outputting it intends to do.

AMD_LOG_LEVEL=4 AMD_LOG_MASK=0x01  python3 <MLworkload_script>.py (should printout api calls)
AMD_LOG_LEVEL=4 AMD_LOG_MASK=0x02 python3 <MLworkload_script>.py (should printout kernel and command barriers only)
AMD_LOG_LEVEL=4 AMD_LOG_MASK=0x2000  python3 <MLworkload_script>.py (should printout AQL packet in raw)

https://github.com/ROCm-Developer-Tools/HIP/blob/develop/docs/developer_guide/logging.md

Regardless of mask value 0x01, 0x02 or 0x2000, output is always same: 

```
:3:                         :0   : 570870393265 us: 35281: [tid:0x7f0c7f813000] HMM support: 1, xnack: 0, direct host access: 0

:3:                         :0   : 570870394238 us: 35281: [tid:0x7f0c7f813000] HMM support: 1, xnack: 0, direct host access: 0

:3:                         :0   : 570870395091 us: 35281: [tid:0x7f0c7f813000] HMM support: 1, xnack: 0, direct host access: 0

:3:                         :0   : 570870395991 us: 35281: [tid:0x7f0c7f813000] HMM support: 1, xnack: 0, direct host access: 0

:3:                         :0   : 570870396908 us: 35281: [tid:0x7f0c7f813000] Direct Dispatch: 1
:3:                         :0   : 570870679323 us: 35281: [tid:0x7f0c7f813000] For Init/Fini: Kernel Name: __amd_rocclr_copyBufferAligned
:3:                         :0   : 570870679338 us: 35281: [tid:0x7f0c7f813000] For Init/Fini: Kernel Name: __amd_rocclr_copyBufferRect
:3:                         :0   : 570870679341 us: 35281: [tid:0x7f0c7f813000] For Init/Fini: Kernel Name: __amd_rocclr_copyBufferRectAligned
:3:                         :0   : 570870679343 us: 35281: [tid:0x7f0c7f813000] For Init/Fini: Kernel Name: __amd_rocclr_streamOpsWait
:3:                         :0   : 570870679345 us: 35281: [tid:0x7f0c7f813000] For Init/Fini: Kernel Name: __amd_rocclr_copyImage
:3:                         :0   : 570870679346 us: 35281: [tid:0x7f0c7f813000] For Init/Fini: Kernel Name: __amd_rocclr_fillBufferAligned
:3:                         :0   : 570870679348 us: 35281: [tid:0x7f0c7f813000] For Init/Fini: Kernel Name: __amd_rocclr_copyImage1DA
:3:                         :0   : 570870679350 us: 35281: [tid:0x7f0c7f813000] For Init/Fini: Kernel Name: __amd_rocclr_copyImageToBuffer
:3:                         :0   : 570870679352 us: 35281: [tid:0x7f0c7f813000] For Init/Fini: Kernel Name: __amd_rocclr_streamOpsWrite
:3:                         :0   : 570870679353 us: 35281: [tid:0x7f0c7f813000] For Init/Fini: Kernel Name: __amd_rocclr_copyBuffer
:3:                         :0   : 570870679355 us: 35281: [tid:0x7f0c7f813000] For Init/Fini: Kernel Name: __amd_rocclr_fillBufferAligned2D
:3:                         :0   : 570870679357 us: 35281: [tid:0x7f0c7f813000] For Init/Fini: Kernel Name: __amd_rocclr_copyBufferToImage
:3:                         :0   : 570870679359 us: 35281: [tid:0x7f0c7f813000] For Init/Fini: Kernel Name: __amd_rocclr_fillImage
:3:                         :0   : 570870679360 us: 35281: [tid:0x7f0c7f813000] For Init/Fini: Kernel Name: __amd_rocclr_gwsInit
11/08/2023 15:58:32: train entered...
Epoch: 0:
training 0th batch...
training 100th batch...
validating 0th batch...
validating 20th batch...

Epoch 1 accuracy: 0.8224 val_accuracy: 0.9688
Epoch: 1:
training 0th batch...
training 100th batch...
validating 0th batch...
validating 20th batch...

Epoch 2 accuracy: 0.9740 val_accuracy: 0.9764
Epoch: 2:
training 0th batch...
training 100th batch...
validating 0th batch...

```

ML workload py script is below:

```
# x - input matrix, usually a bitmap image.
# w - kernel/filter = smaller moving array over x.
# p - padding, extra spacing around original bitmap image.
# s - stride. Incremental move amount by kernel/filter.
# o - final result, feature map.

import torch
import torchvision
from torchvision import transforms
from torch.utils.data import Subset
from torch.utils.data import DataLoader
import torch.nn as nn
import matplotlib
import matplotlib.pyplot as plt
import code
import numpy as np
#import PyQt5
from functools import wraps
from datetime import datetime
m = nn.Softmax(dim=1)

CONFIG_ENABLE_PLOT=0
CONFIG_ENABLE_GPU=1
DEBUG=0
def print_fcn_name(func):
    @wraps(func)
    def wrapper(*args, **kwargs):
        now = datetime.now()
        dt_string = now.strftime("%d/%m/%Y %H:%M:%S")
        print(f'{dt_string}: {func.__name__} entered...')
        result = func(*args, **kwargs)
        return result

    return wrapper

image_path = "./"

transform = transforms.Compose([transforms.ToTensor()])

mnist_dataset = torchvision.datasets.MNIST(root=image_path, train=True, transform=transform, download=True)
mnist_valid_dataset = Subset(mnist_dataset, torch.arange(10000))
mnist_train_dataset = Subset(mnist_dataset, torch.arange(10000, len(mnist_dataset)))
mnist_test_dataset = torchvision.datasets.MNIST(root=image_path, train=False, transform=transform, download=False)

batch_size = 256
torch.manual_seed(1)
train_dl = DataLoader(mnist_train_dataset, batch_size, shuffle=True)
valid_dl = DataLoader(mnist_valid_dataset, batch_size, shuffle=False)

model=nn.Sequential()
if CONFIG_ENABLE_GPU:
    model.to('cuda')

model.add_module('conv1', nn.Conv2d(in_channels=1, out_channels=32, kernel_size=5, padding=2))
model.add_module('ReLU1', nn.ReLU())
model.add_module('pool1', nn.MaxPool2d(kernel_size=2))
model.add_module('conv2', nn.Conv2d(in_channels=32, out_channels=64, kernel_size=5, padding=2))
model.add_module('ReLU2', nn.ReLU())
model.add_module('pool2', nn.MaxPool2d(kernel_size=2))

model.add_module('flatten', nn.Flatten())
model.add_module('fc1', nn.Linear(3136, 1024))
model.add_module('ReLU3', nn.ReLU())
model.add_module('dropout', nn.Dropout(p=0.5))
model.add_module('fc2', nn.Linear(1024,10))
model.add_module('softmax', nn.Softmax(dim=1))

loss_fn = nn.CrossEntropyLoss()
optimizer = torch.optim.Adam(model.parameters(), lr=0.001)
@print_fcn_name
def train(model, num_epochs, train_dl, valid_dl):
    loss_hist_train = [0] * num_epochs
    accuracy_hist_train = [0] * num_epochs
    loss_hist_valid = [0] * num_epochs
    accuracy_hist_valid = [0] * num_epochs

    for epoch in range(num_epochs):
        print(f'Epoch: {epoch}:')
        counter=0
        for x_batch, y_batch in train_dl:
            if DEBUG or 1:
                if counter % 100 == 0:
                    print(f'training {counter}th batch...')

#           if not counter % 10:
#               print(f'.', end='')
            if CONFIG_ENABLE_GPU:
                x_batch.cuda()
                y_batch.cuda()
            pred=model(x_batch)
            loss=loss_fn(pred, y_batch)
            loss.backward()
            optimizer.step()
            optimizer.zero_grad()
            loss_hist_train[epoch] += loss.item()*y_batch.size(0)
            is_correct = (torch.argmax(pred, dim=1) == y_batch).float()
            accuracy_hist_train[epoch] += is_correct.sum()
            counter+=1

        loss_hist_train[epoch] /= len(train_dl.dataset)
        accuracy_hist_train[epoch] /= len(train_dl.dataset
        model.eval()

        with torch.no_grad():
            counter=0
            for x_batch, y_batch in valid_dl:
#               print(f':', end="")
                if DEBUG or 1:
                    if counter % 20 == 0:
                        print(f'validating {counter}th batch...')
                if CONFIG_ENABLE_GPU:
                    x_batch.cuda()
                    y_batch.cuda()
                pred=model(x_batch)
                loss=loss_fn(pred, y_batch)
                loss_hist_valid[epoch] += \
                    loss.item()*y_batch.size(0)
                is_correct=(torch.argmax(pred, dim=1) == y_batch).float()
                accuracy_hist_valid[epoch] += is_correct.sum()
                counter+=1
        loss_hist_valid[epoch] /= len(valid_dl.dataset)
        accuracy_hist_valid[epoch] /= len(valid_dl.dataset)

        print(f'\nEpoch {epoch+1} accuracy: '
            f'{accuracy_hist_train[epoch]:.4f} val_accuracy: '
            f'{accuracy_hist_valid[epoch]:.4f}')

    return loss_hist_train, loss_hist_valid, accuracy_hist_train, accuracy_hist_valid

#code.interact(local=locals())

torch.manual_seed(1)
num_epochs=3
hist=train(model, num_epochs, train_dl, valid_dl)
if CONFIG_ENABLE_PLOT:
    x_arr = np.arange(len(hist[0])) + 1
    fig = plt.figure(figsize=(12,4))
    ax = fig.add_subplot(1,2,1)
    ax.plot(x_arr, hist[0], '-0', label='Train Loss')
    ax.plot(x_arr, hist[1], '--<', label='Validation Loss')

    ax.legend(fontsize=15)
    ax=fig.add_subplot(1,2,2)
    ax.plot(x_arr, hist[2], '-0', label='Train Acc')
    ax.plot(x_arr, hist[3], '--<', label='Validatin Acc')

    ax.legend(fontsize=15)
    ax.set_xlabel('Epoch', size=15)
    ax.set_ylabel('Accuracy', size=15)
    plt.show()
    plt.plot
    plt.savefig("p474.plot1.jpg", dpi=300)

pred=model(mnist_test_dataset.data.unsqueeze(1) / 255. )
is_correct = (torch.argmax(pred, dim=1) == mnist_test_dataset.targets).float()
print(f'Test accuracy: {is_correct.mean():.4f}')

if CONFIG_ENABLE_PLOT:
    fig=plt.figure(figsize=(12, 4))

    for i in range(12):
        ax=fig.add_subplot(2,6,i+1)
        ax.set_xticks([]); ax.set_yticks([])

        img=mnist_test_dataset[i][0][0,:,:]
        pred=model(img.unsqueeze(0).unsqueeze(1))
        y_pred = torch.argmax(pred)
        ax.imshow(img, cmap='gray_r')
        ax.text(0.9, 0.1, y_pred.item(),
            size=15, color='blue',
            horizontalalignment='center',
            verticalalignment='center',
            transform=ax.transAxes)
    plt.show()

```

**Comment by iassiour** [comment]
From the looks of you command:

AMD_LOG_LEVEL=4 AMD_LOG_MASK=0x01 python3 <MLworkload_script>.py

It seems like you are passing a hex value to the environment variable AMD_LOG_MASK.
I don't believe this will work. Could you try to pass a decimal value instead.

**Comment by jdgh000** [comment]
Decimal values appears to work. It is bit weird that someone has to translate to decimal, instruction published is in hex format. I dont think users will ever figure out (at least some) unless specifically asked for it.

**Comment by iassiour** [comment]
I agree. The document should mention that a conversion is needed or alternatively the mask can be set with integers instead, since it is exposed as an environment variable and its enum appears in the docs especially when used  alongside AMD_LOG_LEVEL that is also an integer. Let me followup on this.

**Comment by iassiour** [comment]
I have added changes in the internal repo to initialise the LogMask enum with decimal values and it will also be published in the documentation as such i.e

enum LogMask {
  LOG_API         = 1,      //!< (0x1)     API call
  LOG_CMD       = 2,      //!< (0x2)     Kernel and Copy Commands and Barriers
  LOG_WAIT      = 4,      //!< (0x4)     Synchronization and waiting for commands to finish
  LOG_AQL        = 8,      //!< (0x8)     Decode and display AQL packets
  ...
};

This will hopefully make the usage more clear. The changes should become available in the rocm 6.0 release. 
Closing this issue.

**Comment by iassiour** [comment]
I have added changes in the internal repo to initialise the LogMask enum with decimal values and it will also be published in the documentation as such i.e

enum LogMask {
LOG_API = 1, //!< (0x1) API call
LOG_CMD = 2, //!< (0x2) Kernel and Copy Commands and Barriers
LOG_WAIT = 4, //!< (0x4) Synchronization and waiting for commands to finish
LOG_AQL = 8, //!< (0x8) Decode and display AQL packets
...
};

This will hopefully make the usage more clear. The changes should become available in the rocm 6.0 release.
Closing this issue.
